reference/conanfile/tools/cmake.rst

```diff
@@ -41,14 +41,74 @@ The full instantiation, that allows custom configuration can be done in the ``ge
 
         def generate(self):
             cmake = CMakeDeps(self)
-            cmake.configurations.append("ReleaseShared")
-            if self.options["hello"].shared:
-                cmake.configuration = "ReleaseShared"
             cmake.generate()
 
-As it can be seen, it allows to define custom user CMake configurations besides the standard Release, Debug, etc ones.
-If the **settings.yml** file is customized to add new configurations to the ``settings.build_type``, then, adding it
-explicitly to ``.configurations`` is not necessary.
+There are some attributes you can adjust in the created ``CMakeDeps`` object to change the default behavior:
+
+configurations
+++++++++++++++
+
+As it can be seen in the following example, it allows to define custom user CMake configurations besides the standard
+Release, Debug, etc ones. If the **settings.yml** file is customized to add new configurations to the
+``settings.build_type``, then, adding it explicitly to ``.configurations`` is not necessary.
+
+.. code-block:: python
+
+    ...
+        cmake = CMakeDeps(self)
+        cmake.configurations.append("ReleaseShared")
+        if self.options["hello"].shared:
+            cmake.configuration = "ReleaseShared"
+        cmake.generate()
+
+
+arch
```

Comment:
I'd hide this at this moment, not fully clear the use case, or how it could derive, might be misused.